pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -976,11 +977,21 @@ func (c *Cloud) NodeAddresses(name types.NodeName) ([]v1.NodeAddress, error) {
 	if c.selfAWSInstance.nodeName == name || len(name) == 0 {
 		addresses := []v1.NodeAddress{}
 
-		internalIP, err := c.metadata.GetMetadata("local-ipv4")
+		macs, err := c.metadata.GetMetadata("network/interfaces/macs/")
 		if err != nil {
-			return nil, fmt.Errorf("error querying AWS metadata for %q: %q", "local-ipv4", err)
+			return nil, fmt.Errorf("error querying AWS metadata for %q: %q", "network/interfaces/macs", err)
+		}
+
+		for _, macID := range strings.Split(macs, "\n") {
+			macPath := path.Join("network/interfaces/macs/", macID, "local-ipv4s")
```

Comment:
Probably good to skip empty strings here just in case, but I don't think it is needed for the EC2 metadata service.